internal/olm/fbcutil/util.go

```diff
@@ -0,0 +1,151 @@
+// Copyright 2022 The Operator-SDK Authors
```

Comment:
What happens here in the following scenarios?

a) By calling the run bundle informing and populate index, it should fail when the bundle version is <= than the head of channels where it should be shipped and exist in the index informed. Is that failing?

b) By calling the upgrade bundle configured to replace/skip or skip range informing a populate index are we testing the upgrade path with the bundles which pre-exist in the index (as the other way around)?

c) By configuring a default channel that is not the package default channel in the index informed is it failing?

It seems to me that the above scenarios will not work. If so, what about we remove the option to use these commands with a populate index now that we are adding FBC catalog support to them and then make it simple by not providing a wrong expectation over it?

@rashmigottipati @VenkatRamaraju could you check this scenario and let us know if it works? 

@J0zi theoretically this PR would fix the issues for the run bundle/upgrade bundle. Could you help us out to test the changes? Note that by getting this PR and running make install we produce a bin for SDK so that we can do the tests indeed with the pipeline scenarios where it fails. 

c/c @joelanford

Reply:
Sure @camilamacedo86, will check these two scenarios and update with the results.

Also, are you suggesting to remove `--index-image` flag altogether for these commands?

Reply:
@camilamacedo86 And regarding the error you mentioned in the [original github issue](https://github.com/operator-framework/operator-sdk/issues/5773), this error is caused is `Error: error loading bundle into db: FOREIGN KEY constraint failed` which is due to "being given a default channel that does not exist in the graph". 

And this cannot be handled on the SDK side as SDK delegates the bundle pulling/adding to index to opm. IIRC, the consensus was that OLM will be able to review/merge a PR that would add more context to this error. So I not exactly sure that we could change anything on the SDK side regarding this. 

Having said that, this PR will be a solution for those cases I think because this won't be calling the deprecated and soon-to-be removed opm commands but rather use FBC by default.

Reply:
Hi @rashmigottipati,

Thank you. Regards: https://github.com/operator-framework/operator-sdk/pull/5809#discussion_r882810138

> Sure @camilamacedo86, will check these two scenarios and update with the results.
> Also, are you suggesting to remove --index-image flag altogether for these commands? 

By informing the index, I as a user would be expected to use the command to properly validate the bundle upgrade path with the data that I provided (the index that I informed) If SDK is unable to do that or it is too complex and by having it we are providing wrong expectations or leading people think that an config is valid when it is not then, I am as a user would prefer have not this option. 

If that is the case so: my suggestion/idea here is could we not remove this option and make things simple?

Reply:
@rashmigottipati,

(ihmo) we could : 

- Add a comment in the flag and explain that it does not work appropriately to check the upgraded graph but is useful to validate the deps.
OR
- Deprecated the flag and say that it no longer and only allows it still used with SQLIndex 
OR
- We can see how we could make it work probably. 

The name is bundle upgrade so that seems more intuitive to assume that it helps me out test the upgraded graph and not only deps when an index is informed. So, that concerns me. However, I think at the minimum we need to know what is or is not possible to do with, what works or not and clarify.

Reply:
I update the first comment to clarify.

Reply:
@rashmigottipati FBC does not have the channels config in the bundle. So, maybe the question here is: 
- Is it technically possible/achievable that we have a command that validates the upgrade with a previous index with FBC? OR what is achievable? 
- How much value vs effort from what is achieved do with a previous index now we have to keep the --index image? 

c/c @joelanford

Reply:
TL;DR
# document flag's deficiency
# create an issue for the problem
# in a followup, fix the above mentioned issue.


IMO, I think the best short term solution is to document the flags deficiencies. I think removing the flag is wrong because we will want to support scenarios where the index is input for dependency purposes. Also, removing it would be a breaking change. Then we should file an issue to fix it for item c above "See how we could make it work probably. (Maybe we can now only ensure that we clarifies its limitations and look for that in the future)"

Reply:
I can agree and go along with the @jmrodri thoughts 
we need to track that then it shows ok for me.
Would be great if we could add that in the --help of the commands as well to WARN the users.

Reply:
I definitely agree that it's good to document the flag's limitations. I have added some notes in the help text of the command to let users know of the flag's shortcomings. We look look into this issue and fix it in follow-ups. Also, just an FYI - this is a known existing issue for SQLite indexes as well and not just for FBC. 

Here's the GH link to track this issue: https://github.com/operator-framework/operator-sdk/issues/5773